### PR TITLE
Add support for nodeless dev setup in dev-setup-admission-azure.sh

### PR DESCRIPTION
**Pull request by stoyanr**
**How to categorize this PR?**

/area dev-productivity
/kind enhancement
/platform azure

**What this PR does / why we need it**:
Adds support for "Nodeless" dev setup in `dev-setup-admission-azure.sh`. 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

Without the proposed change, on the "Nodeless" setup calls to the admission webhook failed with an error such as:
```
$ k edit secret seed-operator-az -n garden
error: secrets "seed-operator-az" could not be patched: Internal error occurred: failed calling webhook "secrets.validation.azure.provider.extensions.gardener.cloud": Post "https://gardener-extension-admission-azure.garden.svc:443/webhooks/validate/secrets?timeout=10s": context deadline exceeded
```

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: hack/dev-setup-admission-azure.sh [hunk 1]
```diff
@@ -18,22 +18,48 @@ set -o errexit
...
-# Make it able to work on minikube and nodeless
+is_nodeless() {
+    nodes_len=$(kubectl get node -o json | jq ".items" | jq '. | length')
+    if [[ "$nodes_len" == "0" ]]; then
+      echo "true"
+      return
+    fi
+
+    echo "false"
+    return
```
Comment (ialidzhikov) [inline hack/dev-setup-admission-azure.sh:29]: I guess the idiomatic bash way is:

```suggestion
    if [[ "$nodes_len" == "0" ]]; then
      return 0
    fi

    return 1
```

and

```bash
if is_nodeless; then

else

fi
```

File: hack/dev-setup-admission-azure.sh [hunk 2]
```diff
@@ -18,22 +18,48 @@ set -o errexit
...
+
 if kubectl -n garden get service "$ADMISSION_SERVICE_NAME" &> /dev/null; then
   kubectl -n garden delete service $ADMISSION_SERVICE_NAME
 fi
 if kubectl -n garden get endpoints "$ADMISSION_ENDPOINT_NAME" &> /dev/null; then
   kubectl -n garden delete endpoints $ADMISSION_ENDPOINT_NAME
 fi
 
-cat <<EOF | kubectl apply -f -
+if [[ "$(is_nodeless)" == "true" ]]; then
```
Comment (ialidzhikov) [inline hack/dev-setup-admission-azure.sh:50]:
```suggestion
if is_nodeless; then
```

**Comment by stoyanr** [comment]
@ialidzhikov Thanks for the suggestions! I simply committed and tested them to make sure everything is ok.
/squash

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**